aiverify-foundation/moonshot-data
# Adding testing files for cyberseceval,mlcommons and updated answercarefully.

## Description

Adding testing files for cyberseceval,mlcommons and updated answercarefully and includes updates for issue 141 

## Motivation and Context

Testing the safety of responses in multiple languages ensures that LLMs remain inclusive, ethical, and reliable for a diverse global audience. It also helps identify and address biases, cultural differences, and technical gaps in training data, leading to better performance and user trust.

## Type of Change

## How to Test

I added four languages ​​of the AnswerCarefullyVersion002_Test dataset in the “datasets” folder, including English AnswerCarefully_en.json), Chinese (AnswerCarefully_ch.json), Japanese (AnswerCarefully_jp.json), and Korean (AnswerCarefully_ko.json).

You can check the safety of answers from three models: Llama 3.1, 405B//Mixtral-8x22B Instruct v0.1//GDM Gemma 2, 27B judged by gpt4o.

Take the English (AnswerCarefully_en.json) for example:
Please run the command line:

run_recipe "AnswerCarefully_en_gemma" "['AnswerCarefully_en']" "['together-gemma2-27b']" -n 10 -r 1 -s " "

Just replacing “AnswerCarefully_en” in the command line with the names of other recipes is ok.

Those are the test results for a subset with only 10 examples. If you need to test on the full dataset, please remove “-n 10” from the command line.

--------------------------------------------------------------------------------------------------------------------------------------------

Cybersec_eval:

run_recipe "cyberseceval-gemma" "['cyberseceval_en']" "['together-gemma2-27b']" -n 10 -r 1 -s " "

--------------------------------------------------------------------------------------------------------------------------------------------

Mlcommons:

run_recipe "mlcipv_en-gemma" "['mlc-ipv_en']" "['my-together-gemma2-27b']" -n 10 -r 1 -s " "
run_recipe "mlcncr_en-gemma" "['mlc-ncr_en']" "['my-together-gemma2-27b']" -n 10 -r 1 -s " "
run_recipe "mlcvcr_en-gemma" "['mlc-vcr_en']" "['my-together-gemma2-27b']" -n 10 -r 1 -s " "
run_recipe "mlcprv_en-gemma" "['mlc-prv_en']" "['my-together-gemma2-27b']" -n 10 -r 1 -s " "


To run the cookbook which has all the 4 recipes for english dataset.
run_cookbook "my new cookbook runner" "['mlc-cookbook']" "['together-gemma2-27b']" -n 10 -r 1 -s ""

## Checklist

Please check all the boxes that apply to this pull request using "x":

- []  I have added appropriate unit tests or functional tests for the changes made.
- []  I have rebased my branch onto the latest commit of the main branch.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
